enhancements/olm/marketplace-api-deprecation.md

```diff
@@ -0,0 +1,107 @@
+- Continue to support configuration for OperatorHub
+
+### Non-Goals
+
+- Preserve non-default OperatorSource and CatalogSourceConfig data
+
+## Proposal
+
+### Removal of the Marketplace APIs
+
+Remove the OperatorSource and CatalogSourceConfig CRDs.
+
+Migrate the default OperatorSources to CatalogSources.
+
+Remove the reconcilation of the OperatorSource and CatalogSourceConfig APIs either by removing the marketplace operator entirely or by removing the relevant portions of it (if it is being retained to manage the OperatorHub API, see below).
+
+### Migration of the OperatorHub Config API
+
+The OperatorHub config API is used to configure which catalogs are available by default in a cluster, and is an important configuration point for disconnected clusters.
+
+This API is currently managed by the marketplace operator and needs to continue to be supported. This could be by continuing to support the marketplace operator as an "operatorhub config operator", by migrating the functionality into OLM, or into a new operator in the OLM repo. The decision here will also determine if we will continue to update a separate ClusterOperator object in the marketplace namespace.
+
+The config API also currently configures OperatorSources, and needs to be changed to configure CatalogSources.
+
+### Risks and Mitigations
+
+It is possible that some customers are heavily using the appregistry-based catalogs and marketplace APIs. Because we believe this use to be minimal at best, we have not provided an automated migration plan for those cases.
+
+If we determine that uninterrupted use of appregistry is desired or required, this proposal will need to include migration steps for all OperatorSource and CatalogSourceConfig APIs in-use in a cluster.
+
+Otherwise, a manual guide for migrating custom catalogs should be straightforward to follow.
+
+## Design Details
+
+### Graduation Criteria / Deprecation Plan
+
+Deprecation of the OperatorSource and CatalogSourceConfig APIs has been communicated in the [4.2 release notes](https://docs.openshift.com/container-platform/4.2/release_notes/ocp-4-2-release-notes.html)
+
+Removal of the marketplace operator will not yet remove the operator-marketplace namespace. This namespace is currently the designated namespace for global catalogs.
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, the default catalog contents (redhat-operators, certified-operators, community-operators) will be converted into image-based CatalogSources that poll for updates.
+
+It should not be necessary to update any Subscriptions, as the new CatalogSources will have the same names as the old ones that were generated off of the OperatorSources (TODO: verify)
+
+The OperatorHub config api may cease to be managed by the marketplace operator, and instead may be managed by OLM operator.
+
+### Version Skew Strategy
+
+During upgrade, there may be a period of time when the OperatorHub config API could be managed by two operators (marketplace and OLM).
```

Comment:
But each operator responds to these APIs by watching/creating different resources. Can you clarify what the issue is? Is it a naming conflict?

Reply:
I think so -- the default `CatalogSources` would be contended.

Reply:
It's not the catalogsources that are the issue, it's the config api itself. 

This is an only an issue if we take the operatorhub api out of marketplace and put it into olm - if we do that, then on upgrade there are two operators owning the api.